### DISCUSSION TITLE
Why does `getChainId` return `number` instead of `string`?

### DISCUSSION THREAD

**Discussion by Arvolear** [ethereum/web3.js#5851] [General]
The web3js [getChainId](https://web3js.readthedocs.io/en/v1.8.2/web3-eth.html#getchainid) function returns `number`, however [EIP-695](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-695.md#abstract) states that the RPC returns `string`. Is this a valid behavior?

**Comment by nikoulai** [comment]
Hi
Thanks for reaching out.

I believe past maintainers thought, `number` was more valuable this way.

I don't think this will change, since it's not a critical issue, and will lead to a breaking change for the library. However, this issue is solved in `4.x` ( by passing the `DataFormat` of your choice for returning values), which, is in `alpha` currently.
